docs/authentication.md

```
@@ -0,0 +1,177 @@
+    G -- no --> H
+    G -- yes --> I{Role among the declared ones?}
+    I -- no --> H
+    I -- yes --> C
+```
+
+There is no implicit default. The allowed set is whatever `@Roles(...)` names, and an absent
+or empty list admits nobody.
+
+## Promoting a guest
+
+```
```

markdownlint-cli2 (0.23.2)

[warning] Fenced code blocks should have a language specified

(MD040, fenced-code-language)